### PR TITLE
Make enum methods `values` and `valueOf` unmappable

### LINKED ISSUE #330: Enum methods are marked as deobfuscated and can be renamed
The methods declared in Enum class or added by the compiler are marked as deobfuscated and can be renamed (like `values()`,  `valueOf()` etc)

### PR DESCRIPTION
**Pull request by NebelNidas**
Taken from https://github.com/QuiltMC/enigma/commit/fe9c232b65beb7e27964a92c1dc55509de4da127, https://github.com/QuiltMC/enigma/commit/1d8067cf6526223a9efa74e83d22addd529a4289 ~~and https://github.com/QuiltMC/enigma/commit/226d5b03b4145b8675b1b5dbffc153eb88e97db1~~.

Fixes #330.

### CODE REVIEW

File: enigma-swing/testProject/src/main/java/tests/quilt/issue10/TestClass.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Class to reproduce issue #10
+ * https://github.com/QuiltMC/enigma/issues/10
+ */
+
+package tests.quilt.issue10;
```
Comment (modmuss50) [inline enigma-swing/testProject/src/main/java/tests/quilt/issue10/TestClass.java:6]: Incorrect package

File: enigma-swing/testProject/src/main/java/tests/quilt/issue10/TestClass.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Class to reproduce issue #10
+ * https://github.com/QuiltMC/enigma/issues/10
```
Comment (modmuss50) [inline enigma-swing/testProject/src/main/java/tests/quilt/issue10/TestClass.java:3]: Link to the one on this repo.

File: enigma-swing/testProject/src/main/java/tests/quilt/issue10/TestClass.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+package tests.quilt.issue10;
+
+import java.util.function.Function;
+
+public class TestClass {
```
Comment (modmuss50) [inline enigma-swing/testProject/src/main/java/tests/quilt/issue10/TestClass.java:10]: Where is this test class used? Is there a unit test for it? What is `testProject`? I dont see how this is even built.

**Review by modmuss50 — CHANGES_REQUESTED** [review]
Idk what this test class is doing?

**Comment by NebelNidas** [comment]
Sorry, this should have remained a draft since there were some issues in my cherry-picking. Force-pushed the rewritten commits now so that we don't have the tests in yet, I'll add those under the correct packages in a later PR

File: enigma/src/main/java/cuchaz/enigma/source/cfr/EnigmaDumper.java [hunk 4]
```diff
@@ -273,7 +273,8 @@ public Dumper methodName(String name, MethodPrototype method, boolean special, b
 		int now = sb.length();
 		Token token = new Token(now - name.length(), now, name);
 
-		if (entry != null) {
+		// Skip constructor references
+		if (entry != null && !name.equals("new")) {
```
Comment (modmuss50) [inline enigma/src/main/java/cuchaz/enigma/source/cfr/EnigmaDumper.java:277]: Is this change related, its not clear whats its doing?

Comment (NebelNidas) [reply enigma/src/main/java/cuchaz/enigma/source/cfr/EnigmaDumper.java:277]: It's intended, see [here](https://github.com/QuiltMC/enigma/issues/10)

**Comment by NebelNidas** [comment]
@liach What do you dislike about this PR? 🤔

**Comment by liach** [comment]
Such preventions are quite pointless; we do't prevent renaming equals, enum entries, record components, or other overrides either

**Comment by NebelNidas** [comment]
Shouldn't Enigma do this though?

**Comment by liach** [comment]
We have cases where we rename an override chain inferable through record component names. IMO marking these gray through name proposal service would be better, especially that these names can inadvertently cause troubles due to proguard works and we wish to rename in such cases

**Comment by NebelNidas** [comment]
OK, I'll close this PR then and commit the constructor remap fix in a new one

**Comment by NebelNidas** [comment]
> ... we do't prevent renaming equals ...

What does this do then?
https://github.com/FabricMC/Enigma/blob/c4a79cf0f3140a8b09a40b42e5b51bb5f5e090f4/enigma/src/main/java/cuchaz/enigma/EnigmaProject.java#L133-L134

Also, while it's not explicitly disallowed to rename `values` and `valueOf` yet, Enigma doesn't support it anyway... So adding the check to `isRenamable` doesn't restrict anyone, instead it would accurately report if Enigma supports renaming the passed element

**Review by modmuss50 — APPROVED** [review]
Change is fine, this could maybe refactored to not allow remapping stuff from Java? Or maybe a larger change where we have a classpath of none re nameable libraries im not sure. Needs some discussion first if we want to do it.
